### PR TITLE
Added SPV decodeHtlcScript RPC

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds SPV decodeHtlcScript RPC

#### Which issue(s) does this PR fixes?:

Fixes part of #202

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/spv/decodeHtlcScript.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  /** Util function to replace a char at index in string */
+  const replaceAt = (s: string, index: number, replacement: string): string => s.substr(0, index) + replacement + s.substr(index + replacement.length)
+
+  it('should createHtlc', async () => {
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/spv/decodeHtlcScript.test.ts:21]:
```suggestion
  it('should decodeHtlcScript', async () => {
```

File: packages/jellyfish-api-core/__tests__/category/spv/decodeHtlcScript.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    const pubKeyA = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const pubKeyB = await container.call('spv_getaddresspubkey', [await container.call('spv_getnewaddress')])
+    const blocks = 10
+    const htlc = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, `${blocks}`])
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/spv/decodeHtlcScript.test.ts:25]:
```suggestion
    const htlc = await container.call('spv_createhtlc', [pubKeyA, pubKeyB, blocks])
```

Comment (Jouzo) [reply packages/jellyfish-api-core/__tests__/category/spv/decodeHtlcScript.test.ts:25]: Getting `DeFiDRpcError: 'JSON value is not a string as expected', code: -1` this way

**Review by surangap — APPROVED** [review]
good to me.

**Comment by canonbrother** [comment]
yea looks great too
FYI, the last approval of `kind/feature` PR is required @fuxingloh 's approval

**Comment by Jouzo** [comment]
> yea looks great too
> FYI, the last approval of `kind/feature` PR is required @fuxingloh 's approval

I am not sure to understand correctly what you mean

**Comment by canonbrother** [comment]
> > yea looks great too
> > FYI, the last approval of `kind/feature` PR is required @fuxingloh 's approval
> 
> I am not sure to understand correctly what you mean

ohh.. just the management stuff, the jellyfish pr will auto merge while getting 2 approvals
surangap took 1
then the last must be fuxing

**Comment by Jouzo** [comment]
> > > yea looks great too
> > > FYI, the last approval of `kind/feature` PR is required @fuxingloh 's approval
> > 
> > 
> > I am not sure to understand correctly what you mean
> 
> ohh.. just the management stuff, the jellyfish pr will auto merge while getting 2 approvals
> surangap took 1
> then the last must be fuxing

Got it ;)

**Review by fuxingloh — APPROVED** [review]

**Comment by fuxingloh** [comment]
> > > > yea looks great too
> > > > FYI, the last approval of `kind/feature` PR is required @fuxingloh 's approval
> > > 
> > > 
> > > I am not sure to understand correctly what you mean
> > 
> > 
> > ohh.. just the management stuff, the jellyfish pr will auto merge while getting 2 approvals
> > surangap took 1
> > then the last must be fuxing
> 
> Got it ;)

Oh yeah, it's gives me a better understanding of `kind/features` going into jellyfish to understanding the evolving nature of the project.

**Pull request merged**